crashtracker/src/collector/receiver_manager.rs

```diff
@@ -128,33 +104,60 @@ impl Receiver {
         .context("Failed to create Unix domain socket pair")
         .map(|(a, b)| (a.into_raw_fd(), b.into_raw_fd()))?;
 
-        // We need to spawn a process without calling atfork handlers, since this is happening
-        // inside of a signal handler.  Moreover, preference is given to
-        // multiplatform-uniform solutions. Although `vfork()` is deprecated, the
-        // alternatives have limitations
-        // * `fork()` calls atfork handlers
-        // * There is no guarantee that `posix_spawn()` will not call `fork()` internally
-        // * `clone()`/`clone3()` are Linux-specific
-        // Accordingly, use `vfork()` for now
-        // NB -- on macos the underlying implementation here is actually `fork()`!  See the top of
-        // this file for details.
-        match unsafe { vfork() } {
+        // See ddcommon::unix_utils for platform-specific comments on alt_fork()
+        match alt_fork() {
             0 => {
                 // Child (noreturn)
-                run_receiver_child(prepared_exec, uds_parent, uds_child, stderr, stdout)
+                let _ = unsafe { libc::close(uds_parent) };
+                run_receiver_child(prepared_exec, uds_child, stderr, stdout)
             }
             pid if pid > 0 => {
                 // Parent
-                let _ = close(uds_child);
-                Ok(Receiver {
-                    receiver_uds: uds_parent,
-                    receiver_pid: pid,
+                let _ = unsafe { libc::close(uds_child) };
+                Ok(WatchedProcess {
+                    uds_fd: uds_parent,
+                    pid,
                     oneshot: true,
                 })
             }
             _ => {
                 // Error
-                Err(anyhow::anyhow!("Failed to fork receiver process"))
+                Err(anyhow::anyhow!("Failed to create receiver process"))
+            }
+        }
+    }
+
+    pub(crate) fn to_collector(
```

Comment:
As we discussed, this should be `Collector::spawn(...) -> Self` where `Collector impl WatchedProcess`

Reply:
👍